src/components/testCaseLanding/common/TestCaseTable/TestCaseTable.tsx

```diff
@@ -0,0 +1,365 @@
+        />
+      ),
+      title: (
+        <TruncateText
+          text={tc.title}
+          maxLength={60}
+          name="title"
+          dataTestId={`test-case-title-${tc.id}`}
+        />
+      ),
+      description: (
+        <TruncateText
+          text={tc.description}
+          maxLength={120}
+          name="description"
+          dataTestId={`test-case-description-${tc.id}`}
+        />
+      ),
+      action: <TestCaseActionButton testCase={tc} handleOpen={handleOpen} />,
+    }));
+
+    return results;
+  };
+
+  type TCRow = {
+    status: any;
+    group: string;
+    title: string;
+    description: string;
+    action: any;
+  };
+  const [data, setData] = useState<TCRow[]>([]);
+  useEffect(() => {
+    if (testCases) {
+      setData(transFormData(testCases));
+    }
+  }, [testCases]);
+  const columns = useMemo<ColumnDef<TCRow>[]>(
+    () => [
+      {
+        header: "Status",
+        cell: (row) => row.renderValue(),
+        accessorKey: "status",
+      },
+      {
+        header: "Group",
+        cell: (row) => row.renderValue(),
+        accessorKey: "group",
+      },
+      {
+        header: "Title",
+        cell: (row) => row.renderValue(),
+        accessorKey: "title",
+      },
+      {
+        header: "Description",
+        cell: (row) => row.renderValue(),
+        accessorKey: "description",
+      },
+      {
+        header: "Action",
+        cell: (row) => row.renderValue(),
+        accessorKey: "action",
+      },
+    ],
+    []
+  );
+
+  const table = useReactTable({
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+  });
+
+  return (
+    <table
+      tw="min-w-full"
+      data-testid="test-case-tbl"
+      className="tcl-table"
+      style={{
+        borderTop: "solid 1px #8c8c8c",
+        borderSpacing: "0 2em !important",
+      }}
+    >
+      <thead tw="bg-slate">
+        {table.getHeaderGroups().map((headerGroup) => (
+          <tr key={headerGroup.id}>
+            {headerGroup.headers.map((header) => (
+              <TH key={header.id} scope="col">
+                {header.isPlaceholder
+                  ? null
+                  : flexRender(
+                      header.column.columnDef.header,
+                      header.getContext()
+                    )}
+              </TH>
+            ))}
+          </tr>
+        ))}
+      </thead>
+      <tbody className="table-body" style={{ padding: 20 }}>
+        {table.getRowModel().rows.map((row) => (
+          <tr
+            key={row.id}
+            className="tcl-tr"
+            data-testid={`test-case-row-${row.id}`}
+          >
+            {row.getVisibleCells().map((cell) => (
+              <td key={cell.id} data-testid={`test-case-title-${cell.id}`}>
+                {flexRender(cell.column.columnDef.cell, cell.getContext())}
+              </td>
+            ))}
+          </tr>
+        ))}
+      </tbody>
+      <Popover
```

Comment:
could we take this opportunity to break out the popover/menu into its own component and reduce the size of the table component?

Reply:
Popover component has been moved to a separate file